client/components/CheckboxList.jsx

```diff
@@ -33,6 +33,8 @@ export default class CheckboxList extends PureComponent {
         this.setState({checkedItems});
         this.informAboutChange(checkedItems);
       }
+    } else if (newProps.checkedItems !== this.props.checkedItems) {
```

Comment:
Though I lean towards this being a good short-term fix, this method is now getting difficult to reason about IMO (in way that lines up with some of the arguments made <a href="https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html">here</a>). A few possibilities to simplify a bit, going from slight change to more radical change:

 1. Switch from `componentWillReceiveProps` (deprecated) to `getDerivedStateFromProps`. Would make intent clearer IMO, but might not be able to fully replace the `this.informAboutChange` part.
 2. Centralize the "all checked" logic in MobX store
 3. Use Hooks (could clean up a lot but biggest change / introduces new paradigm to the app)

(I like idea of trying option 2 personally)

Reply:
I'll let @th0r weigh in on the refactoring opportunity :relaxed:

For now, as this change fixes the bug, let's ship this and consider the refactor in its own PR as it makes it easier to review a refactor when it keeps the existing functionality 1:1 to what was before.